src/ray/core_worker/fiber.h

```diff
@@ -155,7 +165,10 @@ class FiberState {
   /// running at once.
   FiberRateLimiter rate_limiter_;
   /// The fiber event used to notify that all worker fibers are stopped running.
-  FiberEvent fiber_stopped_event_;
+  // This FiberEvent must outlive the FiberState class, because a FiberState instance may
```

Comment:
Maybe something like

```
Since we don't join the fiber_runner_thread, it's possible that the fiber_runner_thread still accesses the fiber_stopped_event after it's deallocated in the main thread. As a result, we use a shared_ptr here to make sure it's not deallocated if fiber_runner_thread still has a reference to it.
```